integration/indexes/indexes_test.rb

```diff
@@ -0,0 +1,181 @@
+# encoding: utf-8
+
+#--
+# Copyright 2013-2016 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require File.dirname(__FILE__) + '/../integration_test_case.rb'
+
+class IndexesTest < IntegrationTestCase
+
+  def setup
+    @@ccm_cluster.setup_schema("CREATE KEYSPACE simplex WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}")
+  end
+
+  # Test for creating indexes
+  #
+  # test_can_create_index tests that indexes can be created using the driver. It first creates a simple table and a
+  # simplex index on that table. It then verifies that the index metadata is being properly retrieved.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-178
+  # @expected_result index should be created and its metadata should be retrieved.
+  #
+  # @test_category indexes
+  #
+  def test_can_create_index
+    cluster = Cassandra.cluster
+    session = cluster.connect('simplex')
+
+    session.execute("CREATE TABLE simplex.test (a text PRIMARY KEY, b text)")
+    session.execute("CREATE INDEX b_index ON simplex.test (b)")
+    sleep(2)
```

Comment:
Use a SchemaChangeListener instead of sleep's to wait for updates. See user_defined_function_test.rb for example.